### PR TITLE
Blog post about grafana ui for loki

### PR DESCRIPTION
**Pull request by davkal**

### CODE REVIEW

File: content/blog/2018-12-28-closer-look-at-grafanas-user-interface-for-loki.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+Without paging, you need to make sure what you are looking for is in the result set of the 1000 rows.
+You do this by filtering.
+The combination of time, labels, and a regular expression should be enough to return a decent result set to scroll through.
+A graph above the log results shows how many log lines coloured by log level for the given query are seen over time.
+If you see a spike, you can use the graph to further zoom in based on patterns you in the graph.
+
+![Filter for log lines in the line histogram.](/assets/img/blog/loki/loki_grafana_filtering.png)
+
+Ultimately, we would like the user to be able to select multiple log streams (just like they can graph data from multiple datasources in Grafana), multiplex them into a single stream, and then allow chained filtering, similar to the classic logging grep use case: `... | grep foo | grep -v bar`.
```
Comment (Dieterbe) [inline content/blog/2018-12-28-closer-look-at-grafanas-user-interface-for-loki.md]: Multiple streams from the same DS or do we want to support multi DS here as well ? Some users may be interested in that

File: content/blog/2018-12-28-closer-look-at-grafanas-user-interface-for-loki.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+Grafana’s user interface for Loki is part of a new Grafana feature called [Explore](http://docs.grafana.org/features/explore/).
+Explore got started with a more query-oriented workflow around troubleshooting with [Prometheus](https://prometheus.io) metrics.
+
+![Metrics and logs side by side](/assets/img/blog/loki/loki_grafana_explore_split.png)
+
+Once you have refined your queries to show the unhealthy behavior, you will likely have identified a service and/or an instance that is acting up.
+This is where Loki comes in.
+Loki uses the same selector logic as Prometheus.
+Thus, using Explore’s Split view, you can have Prometheus and Lok, showing you related metrics and logs side-by-side.
```
Comment (Dieterbe) [inline content/blog/2018-12-28-closer-look-at-grafanas-user-interface-for-loki.md]: Loki typo

File: content/blog/2018-12-28-closer-look-at-grafanas-user-interface-for-loki.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+We’re also making the row limit configurable in the Loki datasource, in case you want to adjust the number according your browser’s performance.
+
+### Explore and Split view
+
+Grafana’s user interface for Loki is part of a new Grafana feature called [Explore](http://docs.grafana.org/features/explore/).
+Explore got started with a more query-oriented workflow around troubleshooting with [Prometheus](https://prometheus.io) metrics.
+
+![Metrics and logs side by side](/assets/img/blog/loki/loki_grafana_explore_split.png)
+
+Once you have refined your queries to show the unhealthy behavior, you will likely have identified a service and/or an instance that is acting up.
```
Comment (Dieterbe) [inline content/blog/2018-12-28-closer-look-at-grafanas-user-interface-for-loki.md]: Let's clarify you mean a Prometheus/metrics query here

File: content/blog/2018-12-28-closer-look-at-grafanas-user-interface-for-loki.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+This is where Loki comes in.
+Loki uses the same selector logic as Prometheus.
+Thus, using Explore’s Split view, you can have Prometheus and Lok, showing you related metrics and logs side-by-side.
+
+### Conclusion
+
+Loki and Grafana are a perfect match.
+The backend is kept lean and space-efficient, while the user interface allows ad-hoc field parsing and simple statistics.
+While we will not be able to answer question like “Top n of X” across a long time range, we still believe that the tradeoff is a very useful one for troubleshooting.
```
Comment (Dieterbe) [inline content/blog/2018-12-28-closer-look-at-grafanas-user-interface-for-loki.md]: We currently are not able but let's not rule that one day we may ?

File: content/blog/2018-12-28-closer-look-at-grafanas-user-interface-for-loki.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
++++
+
+### Introduction
+
+At Grafana Labs we run a plethora of microservices for our hosted offerings.
+We've been very happy with our monitoring with Prometheus, and our distributed tracing with Jaeger.
+But we have had a difficult time finding a log aggregation system that fits our needs.
+Existing solutions offered lots of features, but those same features somehow got in the way of finding logs quickly.
+We decided to build our own simplified log aggregation: [Loki](https://grafana.com/loki).
+It is designed to be lean in the backend.
```
Comment (Dieterbe) [inline content/blog/2018-12-28-closer-look-at-grafanas-user-interface-for-loki.md]: Maybe refer to gouthams blog post / design doc here as well for those who want to learn more about Loki

File: content/blog/2018-12-28-closer-look-at-grafanas-user-interface-for-loki.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+We believe that 1000 rows may just be enough of a data set to run such an analysis on.
+We have implemented a handful of parser to extract fields from log lines, e.g., `instance=foo_1 status=400 ...`.
+This allows us to build dynamic field matchers on demand to gather the distribution of field values across the row set, e.g., `instance` is `foo_1` on 55 % of the rows that have the field `instance`.
+
+![Ad-hoc statistics on a field.](/assets/img/blog/loki/loki_grafana_statistics.png)
+
+Since we are calculating the statistics on the result set that is already in the browser, the results show up instantly.
+It is worth noting that good filtering is important to maximise the number of rows that have the field present that you want to match against.
+We're extending the parse and match logic to also allow drawing of graphs based on number values, e.g., if your log lines contain something line `duration=11.3ms`, this should be graphable as an ad-hoc time series.
```
Comment (bulletfactory) [inline content/blog/2018-12-28-closer-look-at-grafanas-user-interface-for-loki.md]: remove "of"

**Pull request merged**